### PR TITLE
KAFKA-14009: Rebalance timeout should be updated when static member rejoins

### PR DESCRIPTION
**Pull request by Stephan14**
When consumers use static membership protocol and do not happen rebalance, consumers can not update rebalance timeout when users want to reduce rebalance timeout

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by Kvicii — APPROVED** [review]

**Comment by Stephan14** [comment]
@guozhangwang Can you help review this code?

**Comment by dajac** [comment]
@Stephan14 I will review it next week.

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala [hunk 1]
```diff
@@ -1294,7 +1296,7 @@ class GroupCoordinator(val brokerId: Int,
     supportSkippingAssignment: Boolean
   ): Unit = {
     val currentLeader = group.leaderOrNull
-    val member = group.replaceStaticMember(groupInstanceId, oldMemberId, newMemberId)
+    val member = group.replaceStaticMember(groupInstanceId, oldMemberId, newMemberId, rebalanceTimeoutMs)
```
Comment (dajac) [inline core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala]: Did you consider updating the rebalance timeout via the `updateMember` method instead of this one? It feels a bit weird to replace and update the member here.

**Review by dajac — COMMENTED** [review]
@Stephan14 Thanks for the patch. Nice catch! I have two high level comments:
1) Could we add a unit test for this?
2) Do we have a similar issue with the session timeout?

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala (hunk 1 above)
Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala]: ok, I update rebalance timeout and session timeout via the `updateMember`

**Comment by Stephan14** [comment]
> @Stephan14 Thanks for the patch. Nice catch! I have two high level comments:
> 
> 1. Could we add a unit test for this?
> 2. Do we have a similar issue with the session timeout?

@dajac 
1. I have added test for this
2. I update session timeout at the same time

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala [hunk 2]
```diff
@@ -963,6 +963,31 @@ class GroupMetadataManagerTest {
     assertEquals(Some(staticMemberId), group.currentStaticMemberId(groupInstanceId))
   }
 
+  @Test
+  def testupdateMemberWithStaticMember(): Unit = {
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: Would it be more appropriate to have this test in `GroupMetadataTest`?

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala [hunk 3]
```diff
@@ -1300,7 +1304,9 @@ class GroupCoordinator(val brokerId: Int,
     completeAndScheduleNextHeartbeatExpiration(group, member)
 
     val knownStaticMember = group.get(newMemberId)
-    group.updateMember(knownStaticMember, protocols, responseCallback)
+    val oldRebalanceTimeoutMs = knownStaticMember.rebalanceTimeoutMs
+    val oldSessionTimeoutMs = knownStaticMember.sessionTimeoutMs
+    group.updateMember(knownStaticMember, protocols, rebalanceTimeoutMs, sessionTimeoutMs, responseCallback)
```
Comment (dajac) [inline core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:1317]: Could we add a test in `GroupCoordinatorTest` to exercise this path and the failed path as well?

**Review by dajac — COMMENTED** [review]
Thanks for the update. I left a few more comments.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala (hunk 2 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: I have move it to `GroupMetadataTest`

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala (hunk 3 above)
Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:1317]: I can not mock the failed case in test

Comment (dajac) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:1317]: @Stephan14 Could you add a unit test in `GroupCoordinatorTest`? If you have trouble with this, could you explain your issues? I can perhaps help you.

Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:1317]: I have add unit test about persistence failed after joining group successfully in `GroupCoordinatorTest`

Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:1317]: @dajac Can you review it?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 4]
```diff
@@ -1036,6 +1036,28 @@ class GroupCoordinatorTest {
     assertEquals(Errors.NONE, syncGroupWithOldMemberIdResult.error)
   }
 
+ @Test
+ def staticMemberJoinStorageWithUnknownPersistenceFailure(): Unit = {
+   val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId)
+   val joinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, followerInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 *DefaultSessionTimeout, appendRecordError = Errors.MESSAGE_TOO_LARGE)
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Should the second `DefaultSessionTimeout` be `DefaultRebalanceTimeout`? Moreover, a space is missing after `*`.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 5]
```diff
@@ -1036,6 +1036,28 @@ class GroupCoordinatorTest {
...
+     rebalanceResult.generation,
+     Set.empty,
+     Stable,
+     Some(protocolType))
+   assertNotEquals(None, groupCoordinator.groupManager.getGroup(groupId))
+   val group = groupCoordinator.groupManager.getGroup(groupId).get
+   val it = group.allMemberMetadata.iterator
+   while (it.hasNext) {
+     val member = it.next()
+     assertNotEquals(member.sessionTimeoutMs, 2 * DefaultSessionTimeout)
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: I guess that we could remove the `assertNotEquals` assertions, no?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 6]
```diff
@@ -1036,6 +1036,28 @@ class GroupCoordinatorTest {
...
+ def staticMemberJoinStorageWithUnknownPersistenceFailure(): Unit = {
+   val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId)
+   val joinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, followerInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 *DefaultSessionTimeout, appendRecordError = Errors.MESSAGE_TOO_LARGE)
+   checkJoinGroupResult(joinGroupResult,
+     Errors.UNKNOWN_SERVER_ERROR,
+     rebalanceResult.generation,
+     Set.empty,
+     Stable,
+     Some(protocolType))
+   assertNotEquals(None, groupCoordinator.groupManager.getGroup(groupId))
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: Should we use `assertTrue(groupCoordinator.groupManager.getGroup(groupId).isDefined)`?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 7]
```diff
@@ -1036,6 +1036,28 @@ class GroupCoordinatorTest {
...
+   val it = group.allMemberMetadata.iterator
+   while (it.hasNext) {
+     val member = it.next()
+     assertNotEquals(member.sessionTimeoutMs, 2 * DefaultSessionTimeout)
+     assertEquals(member.sessionTimeoutMs, DefaultSessionTimeout)
+     assertNotEquals(member.rebalanceTimeoutMs, 2 * DefaultRebalanceTimeout)
+     assertEquals(member.rebalanceTimeoutMs, DefaultRebalanceTimeout)
+   }
+ }
+
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala:1056]: Could we also add a unit test which verifies that the happy case? I mean, when persisting the change succeeds.

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala [hunk 8]
```diff
@@ -438,7 +440,7 @@ class GroupCoordinator(val brokerId: Int,
           case None => group.currentState match {
             case PreparingRebalance =>
               val member = group.get(memberId)
-              updateMemberAndRebalance(group, member, protocols, s"Member ${member.memberId} joining group during ${group.currentState}; client reason: $reason", responseCallback)
+              updateMemberAndRebalance(group, member, protocols, rebalanceTimeoutMs, sessionTimeoutMs, s"Member ${member.memberId} joining group during ${group.currentState}; client reason: $reason", responseCallback)
```
Comment (dajac) [inline core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:443]: Updating the rebalance timeout and the session timeout in the non-static cases is not really necessary. I suppose that it does not hurt to do it.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala [hunk 9]
```diff
@@ -963,6 +963,8 @@ class GroupMetadataManagerTest {
     assertEquals(Some(staticMemberId), group.currentStaticMemberId(groupInstanceId))
   }
 
+
+
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: nit: Could we revert this change?

**Review by dajac — COMMENTED** [review]
@Stephan14 Please excuse me for the delay on this one. I was on vacation for a while. I just made another pass over it and I left a few comments. Thanks.

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala (hunk 8 above)
Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala:443]: Yes

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 4 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Oh, I fix it

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 6 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Fix it

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 5 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Ok

File: core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala (hunk 9 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala]: Revert it

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 7 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala:1056]: I Have add a test `staticMemberJoinStorageWithUnknownPersistenceSuccess`

**Comment by Stephan14** [comment]
@dajac I have modify the code.

**Comment by Stephan14** [comment]
@dajac Can you help me review it? Thanks.

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala [hunk 10]
```diff
@@ -1308,7 +1312,9 @@ class GroupCoordinator(val brokerId: Int,
     completeAndScheduleNextHeartbeatExpiration(group, member)
 
     val knownStaticMember = group.get(newMemberId)
-    group.updateMember(knownStaticMember, protocols, responseCallback)
+    val oldRebalanceTimeoutMs = knownStaticMember.rebalanceTimeoutMs
+    val oldSessionTimeoutMs = knownStaticMember.sessionTimeoutMs
+    group.updateMember(knownStaticMember, protocols, rebalanceTimeoutMs, sessionTimeoutMs, responseCallback)
     val oldProtocols = knownStaticMember.supportedProtocols
```
Comment (dajac) [inline core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala]: Not related to your PR, but it seems that we should get the old protocols before updating the member, no? We may have a gap in our testing here.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 11]
```diff
@@ -1036,6 +1036,48 @@ class GroupCoordinatorTest {
     assertEquals(Errors.NONE, syncGroupWithOldMemberIdResult.error)
   }
 
+ @Test
+ def staticMemberJoinStorageWithUnknownPersistenceFailure(): Unit = {
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: The test name is not so good here. How about `staticMemberRejoinWithUpdatedSessionAndRebalanceTimeoutsButCannotPersistChange`?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 12]
```diff
@@ -1036,6 +1036,48 @@ class GroupCoordinatorTest {
...
+     Errors.UNKNOWN_SERVER_ERROR,
+     rebalanceResult.generation,
+     Set.empty,
+     Stable,
+     Some(protocolType))
+   assertTrue(groupCoordinator.groupManager.getGroup(groupId).isDefined)
+   val group = groupCoordinator.groupManager.getGroup(groupId).get
+   val it = group.allMemberMetadata.iterator
+   while (it.hasNext) {
+     val member = it.next()
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: Should we use the following?

```
group.allMemberMetadata.foreach { member =>
  assert....
}
```

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 13]
```diff
@@ -1036,6 +1036,48 @@ class GroupCoordinatorTest {
...
+      Set.empty,
+      Stable,
+      Some(protocolType))
+    val leaderJoinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, leaderInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 * DefaultRebalanceTimeout)
+    assertEquals(leaderJoinGroupResult.error, Errors.NONE)
+    assertTrue(groupCoordinator.groupManager.getGroup(groupId).isDefined)
+    val group = groupCoordinator.groupManager.getGroup(groupId).get
+    val it = group.allMemberMetadata.iterator
+    while (it.hasNext) {
+      val member = it.next()
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: same comment as the previous one.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 14]
```diff
@@ -1036,6 +1036,48 @@ class GroupCoordinatorTest {
...
+    val rebalanceResult = staticMembersJoinAndRebalance(leaderInstanceId, followerInstanceId)
+    val followerJoinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, followerInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 * DefaultRebalanceTimeout)
+    checkJoinGroupResult(followerJoinGroupResult,
+      Errors.NONE,
+      rebalanceResult.generation,
+      Set.empty,
+      Stable,
+      Some(protocolType))
+    val leaderJoinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, leaderInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 * DefaultRebalanceTimeout)
+    assertEquals(leaderJoinGroupResult.error, Errors.NONE)
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: Why do you use `checkJoinGroupResult` here as well?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 15]
```diff
@@ -1036,6 +1036,48 @@ class GroupCoordinatorTest {
...
+   while (it.hasNext) {
+     val member = it.next()
+     assertEquals(member.sessionTimeoutMs, DefaultSessionTimeout)
+     assertEquals(member.rebalanceTimeoutMs, DefaultRebalanceTimeout)
+   }
+ }
+
+
+  @Test
+  def staticMemberJoinStorageWithUnknownPersistenceSuccess(): Unit = {
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: Could we also rename this one? The name does not really explain what we are testing here.

**Review by dajac — COMMENTED** [review]
@Stephan14 Excuse me for the delay... Thanks for the update. The PR looks pretty good. I left a few suggestions for consideration.

File: core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala (hunk 10 above)
Comment (Stephan14) [reply core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala]: Fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 11 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 12 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 14 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 13 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 15 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

**Comment by Stephan14** [comment]
@dajac I have fixed it again and can you help me review it ?

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 16]
```diff
@@ -1036,6 +1036,52 @@ class GroupCoordinatorTest {
...
+      Errors.NONE,
+      rebalanceResult.generation,
+      Set.empty,
+      Stable,
+      Some(protocolType))
+    val leaderJoinGroupResult = staticJoinGroupWithPersistence(groupId, JoinGroupRequest.UNKNOWN_MEMBER_ID, leaderInstanceId, protocolType, protocolSuperset, clockAdvance = 1, 2 * DefaultSessionTimeout, 2 * DefaultRebalanceTimeout)
+    checkJoinGroupResult(leaderJoinGroupResult,
+      Errors.NONE,
+      rebalanceResult.generation,
+      Set[String](leaderInstanceId, followerInstanceId),
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: `[String]` is not necessary here. The scala compiler should be able to infer this.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala [hunk 17]
```diff
@@ -1036,6 +1036,52 @@ class GroupCoordinatorTest {
...
+      Stable,
+      Some(protocolType),
+      leaderJoinGroupResult.leaderId,
+      leaderJoinGroupResult.memberId,
+      true)
+    assertTrue(groupCoordinator.groupManager.getGroup(groupId).isDefined)
+    val group = groupCoordinator.groupManager.getGroup(groupId).get
+    group.allMemberMetadata.foreach{ member =>
```
Comment (dajac) [inline core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: nit: A space is missing before `{`.

**Review by dajac — COMMENTED** [review]
@Stephan14 Thanks for the update. I left two really small nits and we should be able to merge the PR.

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 17 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

File: core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala (hunk 16 above)
Comment (Stephan14) [reply core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala]: fixed

**Comment by Stephan14** [comment]
@dajac I have fix it and can you review it?

**Review by dajac — APPROVED** [review]
LGTM, thanks for the patch! I will merge it once the builds succeed.

**Comment by dajac** [comment]
@Stephan14 Many tests, related to your change, failed in the last build. Could you please take a look?

```
Build / JDK 11 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
<1s
Build / JDK 11 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
4s
Build / JDK 8 and Scala 2.12 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
<1s
Build / JDK 8 and Scala 2.12 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
3s
Build / JDK 17 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
<1s
Build / JDK 17 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
```

**Comment by Stephan14** [comment]
> @Stephan14 Many tests, related to your change, failed in the last build. Could you please take a look?
> 
> ```
> Build / JDK 11 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> <1s
> Build / JDK 11 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> 4s
> Build / JDK 8 and Scala 2.12 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> <1s
> Build / JDK 8 and Scala 2.12 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> 3s
> Build / JDK 17 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> <1s
> Build / JDK 17 and Scala 2.13 / staticMemberRejoinWithUnknownMemberIdAndChangeOfProtocolWhileSelectProtocolUnchangedPersistenceFailure() – kafka.coordinator.group.GroupCoordinatorTest
> ```

@dajac  I have fixed it.

**Review by dajac — APPROVED** [review]
LGTM, thanks.

**Comment by dajac** [comment]
@Stephan14 Thanks for the patch!

**Comment by dajac** [comment]
Merged to trunk and 3.3.
